crates/polars-ops/src/series/ops/replace.rs

```diff
@@ -33,7 +33,20 @@ pub fn replace(
         return Ok(default);
     }
 
-    let old = old.strict_cast(s.dtype())?;
+    let old = match (s.dtype(), old.dtype()) {
+        (s_dt, old_dt) if s_dt == old_dt => old.clone(),
+        #[cfg(feature = "dtype-categorical")]
+        (DataType::Categorical(opt_rev_map, ord), DataType::Utf8) => {
```

Comment:
I imagine this could also work when `old` is a (local) Categorical rather than a `Utf8`?

Reply:
For Categorical -> Enum conversion you mean?

For Utf8, I think it is fair to cast to a Categorical / Enum in the background. However, casting from Categorical to ENUM, I don't think it is a good idea to mix & match these types in the background

Reply:
Nevermind, I was thinking the following should work, but it already does on the main branch!

```python
s = pl.Series(["a", "b"], dtype=pl.Categorical)
old = pl.Series(["a"], dtype=pl.Categorical)
new = pl.Series(["c"], dtype=pl.Categorical)

result = s.replace(old, new)
```
```
/home/stijn/code/polars/py-polars/repro.py:7: CategoricalRemappingWarning: Local categoricals have different encodings, expensive re-encoding is done to perform this merge operation. Consider using a StringCache or an Enum type if the categories are known in advance
  result = s.replace(old, new)
shape: (2,)
Series: '' [cat]
[
        "c"
        "b"
]
```